### PR TITLE
[action] Consolidate git_branch action logic into git_helper.rb to ensure consistent results

### LINKED ISSUE #15034: Update git_branch action to support all known environment variables
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When running the following lane in bitrise.io in PR mode, I get different responses from the `git_branch()` action and the internal `Actions.git_branch` action used by the `ensure_git_branch()` action.  
```ruby
platform :android do

  desc "Test both git_branch actions"
  lane :test_git_branch do |options|
    UI.message "Call with Actions.git_branch #{Actions.git_branch}"
    UI.message "Call with git_branch() #{git_branch()}"
  end

end
```

As previously mentioned in the #13346 the `Actions.git_branch` internal action `/fastlane/lib/fastlane/helper/git_helper.rb` should be updated to account for the full range of known CI environment variables to ensure consistency.  Given it's an overloaded method name, it's not immediately clear there are 2 different implementations, as the documentation only references the public action.  The public `git_branch()`action implemented here `/fastlane/lib/fastlane/actions/git_branch.rb` should directly call the internal version, once updated, to keep this logic DRY and provide convenience for those using this action.

##### Command executed

`bundle exec fastlane test_git_branch`
##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> [04:28:07]: [32m------------------------------[0m
[04:28:07]: [32m--- Step: default_platform ---[0m
[04:28:07]: [32m------------------------------[0m
[04:28:07]: [32mDriving the lane 'android test_git_branch' 🚀[0m
[04:28:07]: Call with Actions.git_branch HEAD
[04:28:07]: [32m------------------------[0m
[04:28:07]: [32m--- Step: git_branch ---[0m
[04:28:07]: [32m------------------------[0m
[04:28:07]: Call with git_branch() chore/my-test-branch

+------+------------------+-------------+
|           [32mfastlane summary[0m            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
| 2    | git_branch       | 0           |
+------+------------------+-------------+

[04:28:07]: [32mfastlane.tools finished successfully 🎉[0m

#######################################################################
# fastlane 2.127.1 is available. You are on 2.127.0.
# You should use the latest version.
# Please update using `bundle update fastlane`.
#######################################################################

[32m2.127.1 Improvements[0m
* [fastlane] fix FastlaneCore::BuildWatchr with erroring on multiple builds (#15014) via Josh Holtz
* [actions] fix `SharedValues` warnings (#15013) via Daniel Jankowski

[32mPlease update using `bundle update fastlane`[0m
|                                                                              |
+---+---------------------------------------------------------------+----------+
| [32;1m✓[0m | [32;1mfastlane test_git_branch[0m                                      | 7.94 sec |
+---+---------------------------------------------------------------+----------+
</pre>
</details>

### Environment

<details>
  <pre> <details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | debian_based stretch/sid                                       |
| Ruby                        | 2.5.1                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.7.4                                              |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Ubuntu 16.04.6 LTS debian_based stretch/sid (4.4.0-57-generic) |
| Ruby Lib Dir                | /usr/local/lib                                                 |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                                     |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
default_platform(:android)


platform :android do

  desc "Tests versioning and templating features"
  lane :test_git_branch do |options|
    UI.message "Call with Actions.git_branch #{Actions.git_branch}"
    UI.message "Call with git_branch() #{git_branch()}"
  end

end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
 #redacted

```
</details>

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.127.0 | 🚫 Update available |

### Loaded fastlane plugins:

| Plugin                                      | Version | Update-Status  |
| ------------------------------------------- | ------- | -------------- |
| fastlane-plugin-property_file_read          | 1.0.1   | ✅ Up-To-Date   |
| fastlane-plugin-semantic_release            | 1.1.6   | ✅ Up-To-Date   |
| fastlane-plugin-versioning_android          | 0.1.0   | ✅ Up-To-Date   |

<details><summary><b>Loaded gems</b></summary>

| Gem                                         | Version      |
| ------------------------------------------- | ------------ |
| did_you_mean                                | 1.2.0        |
| bundler                                     | 2.0.2        |
| fileutils                                   | 1.0.2        |
| CFPropertyList                              | 3.0.0        |
| public_suffix                               | 2.0.5        |
| addressable                                 | 2.6.0        |
| atomos                                      | 0.1.3        |
| babosa                                      | 1.0.2        |
| claide                                      | 1.0.2        |
| colored                                     | 1.2          |
| colored2                                    | 3.1.2        |
| highline                                    | 1.7.10       |
| commander-fastlane                          | 4.4.6        |
| declarative                                 | 0.0.10       |
| declarative-option                          | 0.1.0        |
| digest-crc                                  | 0.4.1        |
| unf_ext                                     | 0.0.7.6      |
| unf                                         | 0.1.4        |
| domain_name                                 | 0.5.20190701 |
| dotenv                                      | 2.7.4        |
| emoji_regex                                 | 1.0.1        |
| excon                                       | 0.64.0       |
| multipart-post                              | 2.0.0        |
| faraday                                     | 0.15.4       |
| http-cookie                                 | 1.0.3        |
| faraday-cookie_jar                          | 0.0.6        |
| faraday_middleware                          | 0.13.1       |
| fastimage                                   | 2.1.5        |
| gh_inspector                                | 1.1.3        |
| jwt                                         | 2.1.0        |
| memoist                                     | 0.16.0       |
| multi_json                                  | 1.13.1       |
| os                                          | 1.0.1        |
| signet                                      | 0.11.0       |
| googleauth                                  | 0.6.7        |
| httpclient                                  | 2.8.3        |
| mime-types-data                             | 3.2019.0331  |
| mime-types                                  | 3.2.2        |
| uber                                        | 0.1.0        |
| representable                               | 3.0.4        |
| retriable                                   | 3.1.2        |
| google-api-client                           | 0.23.9       |
| google-cloud-env                            | 1.2.0        |
| google-cloud-core                           | 1.3.0        |
| google-cloud-storage                        | 1.16.0       |
| json                                        | 2.2.0        |
| mini_magick                                 | 4.5.1        |
| multi_xml                                   | 0.6.0        |
| plist                                       | 3.5.0        |
| rubyzip                                     | 1.2.3        |
| security                                    | 0.1.3        |
| naturally                                   | 2.2.0        |
| simctl                                      | 1.6.5        |
| slack-notifier                              | 2.3.2        |
| terminal-notifier                           | 2.0.0        |
| unicode-display_width                       | 1.6.0        |
| terminal-table                              | 1.8.0        |
| tty-screen                                  | 0.7.0        |
| tty-cursor                                  | 0.7.0        |
| tty-spinner                                 | 0.9.1        |
| word_wrap                                   | 1.0.0        |
| nanaimo                                     | 0.2.6        |
| xcodeproj                                   | 1.11.0       |
| rouge                                       | 2.0.7        |
| xcpretty                                    | 0.3.0        |
| xcpretty-travis-formatter                   | 1.0.0        |
| fastlane-plugin-property_file_read          | 1.0.1        |
| fastlane-plugin-semantic_release            | 1.1.6        |
| fastlane-plugin-versioning_android          | 0.1.0        |
</details>

*generated on:* **2019-07-17**</pre>
</details>

### PR DESCRIPTION
**Pull request by arri-cc**
Fixes: #15034 

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

As reported in #15034, in certain scenarios the `git_branch()` Action would return different results than the `git_helper.rb` method with the same name `Actions.git_branch`.  This was discovered while trouble shooting issues with the `ensure_git_branch()` action.  Instead of maintaining 2 copies of the logic to retrieve the current branch, we should consolidate it into the appropriate helper class.

### Description

I simply copied the shared values and logic from `fastlane/lib/fastlane/actions/git_branch.rb` into `fastlane/lib/fastlane/helper/git_helper.rb` and updated the origin to call the updated internal method.

### CODE REVIEW

**Comment by arri-cc** [comment]
I signed it!

**Comment by arri-cc** [comment]
I'll work on resolving the build errors

**Comment by janpio** [comment]
Let me know if you have any problem with that, and I can help you look into how your changes influence the failing tests.

Ping @mollyIV regarding the moving of the `module SharedValues` - this probably doesn't work and should stay where it currently is I guess?

@arri-cc Do you know why we need `git_branch` in the helper anyway? Can we maybe fully move this to the action?

**Comment by arri-cc** [comment]
@janpio it appears moving all of the logic and shared values from the `git_branch` action to the `git_helper.rb` class has broken the tests, while also altering the expected behavior as the convention for the helper class is to return `nil`.  tbh being new to this project I'm not sure why we have actions for git *and* a git helper class.  It would've been easier to just have `ensure_git_branch` call the `git_branch` action instead of the `Actions.git_branch` helper method.

When I run `bundle exec rake test_all` there are failed tests related to github PRs and slack, I'm not sure what to do about those as they seem unrelated when i look at the specs.
```
Failures:

  1) Fastlane Fastlane::FastFile create_pull_request successful correctly submits to github
     Failure/Error:
       result = Fastlane::FastFile.new.parse("
         lane :test do
           create_pull_request(
             api_token: '123456789',
             title: 'test PR',
             repo: 'fastlane/fastlane',
           )
         end
       ").runner.execute(:test)
     
     WebMock::NetConnectNotAllowedError:
       Real HTTP connections are disabled. Unregistered request: POST https://api.github.com/repos/fastlane/fastlane/pulls with body '{"title":"test PR","head":null,"base":"master"}' with headers {'Accept'=>'*/*', 'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'}
     
       You can stub this request with the following snippet:
     
       stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/pulls").
         with(:body => "{\"title\":\"test PR\",\"head\":null,\"base\":\"master\"}",
              :headers => {'Accept'=>'*/*', 'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'}).
         to_return(:status => 200, :body => "", :headers => {})
     
       registered request stubs:
     
       stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/issues/1347/assignees").
         with(:body => "{\"assignees\":[\"octocat\",\"hubot\",\"other_user\"]}",
              :headers => {'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'})
       stub_request(:patch, "https://api.github.com/repos/fastlane/fastlane/issues/1347").
         with(:body => "{\"labels\":[\"fastlane\",\"is\",\"awesome\"]}",
              :headers => {'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'})
       stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/pulls").
         with(:body => "{\"title\":\"test PR\",\"head\":\"git rev-parse --abbrev-ref HEAD\",\"base\":\"master\"}",
              :headers => {'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'})
       stub_request(:any, "/developer.apple.com\/services-account\/QH65B2\/account\/ios\/identifiers\/.*OMC(s){0,1}\.action/")
       stub_request(:any, "/developer.apple.com\/services-account\/QH65B2\/account\/auth\/key/")
       stub_request(:any, "/appstoreconnect.apple.com\/testflight\/v2/")
     
       ============================================================
     # /Library/Ruby/Gems/2.3.0/gems/webmock-2.3.2/lib/webmock/http_lib_adapters/excon_adapter.rb:68:in `handle_request'
     # /Library/Ruby/Gems/2.3.0/gems/webmock-2.3.2/lib/webmock/http_lib_adapters/excon_adapter.rb:40:in `block in add_excon_stub'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/mock.rb:38:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/instrumentor.rb:34:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/idempotent.rb:19:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/base.rb:22:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/base.rb:22:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/connection.rb:270:in `request'
     # ./fastlane/lib/fastlane/actions/github_api.rb:250:in `call_endpoint'
     # ./fastlane/lib/fastlane/actions/github_api.rb:21:in `run'
     # ./fastlane/lib/fastlane/actions/create_pull_request.rb:18:in `run'
     # ./fastlane/lib/fastlane/runner.rb:256:in `block (2 levels) in execute_action'
     # ./fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
     # ./fastlane/lib/fastlane/runner.rb:235:in `block in execute_action'
     # ./fastlane/lib/fastlane/runner.rb:227:in `chdir'
     # ./fastlane/lib/fastlane/runner.rb:227:in `execute_action'
     # ./fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
     # ./fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
     # (eval):3:in `block in parsing_binding'
     # ./fastlane/lib/fastlane/lane.rb:33:in `call'
     # ./fastlane/lib/fastlane/runner.rb:49:in `block in execute'
     # ./fastlane/lib/fastlane/runner.rb:45:in `chdir'
     # ./fastlane/lib/fastlane/runner.rb:45:in `execute'
     # ./fastlane/spec/actions_specs/create_pull_request_spec.rb:49:in `block (5 levels) in <top (required)>'

  2) Fastlane Fastlane::FastFile create_pull_request successful correctly submits to github with labels
     Failure/Error:
       result = Fastlane::FastFile.new.parse("
         lane :test do
           create_pull_request(
             api_token: '123456789',
             title: 'test PR',
             repo: 'fastlane/fastlane',
             labels: ['fastlane', 'is', 'awesome']
           )
         end
       ").runner.execute(:test)
     
     WebMock::NetConnectNotAllowedError:
       Real HTTP connections are disabled. Unregistered request: POST https://api.github.com/repos/fastlane/fastlane/pulls with body '{"title":"test PR","head":null,"base":"master"}' with headers {'Accept'=>'*/*', 'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'}
     
       You can stub this request with the following snippet:
     
       stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/pulls").
         with(:body => "{\"title\":\"test PR\",\"head\":null,\"base\":\"master\"}",
              :headers => {'Accept'=>'*/*', 'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'}).
         to_return(:status => 200, :body => "", :headers => {})
     
       registered request stubs:
     
       stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/issues/1347/assignees").
         with(:body => "{\"assignees\":[\"octocat\",\"hubot\",\"other_user\"]}",
              :headers => {'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'})
       stub_request(:patch, "https://api.github.com/repos/fastlane/fastlane/issues/1347").
         with(:body => "{\"labels\":[\"fastlane\",\"is\",\"awesome\"]}",
              :headers => {'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'})
       stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/pulls").
         with(:body => "{\"title\":\"test PR\",\"head\":\"git rev-parse --abbrev-ref HEAD\",\"base\":\"master\"}",
              :headers => {'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'})
       stub_request(:any, "/developer.apple.com\/services-account\/QH65B2\/account\/ios\/identifiers\/.*OMC(s){0,1}\.action/")
       stub_request(:any, "/developer.apple.com\/services-account\/QH65B2\/account\/auth\/key/")
       stub_request(:any, "/appstoreconnect.apple.com\/testflight\/v2/")
     
       ============================================================
     # /Library/Ruby/Gems/2.3.0/gems/webmock-2.3.2/lib/webmock/http_lib_adapters/excon_adapter.rb:68:in `handle_request'
     # /Library/Ruby/Gems/2.3.0/gems/webmock-2.3.2/lib/webmock/http_lib_adapters/excon_adapter.rb:40:in `block in add_excon_stub'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/mock.rb:38:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/instrumentor.rb:34:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/idempotent.rb:19:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/base.rb:22:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/base.rb:22:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/connection.rb:270:in `request'
     # ./fastlane/lib/fastlane/actions/github_api.rb:250:in `call_endpoint'
     # ./fastlane/lib/fastlane/actions/github_api.rb:21:in `run'
     # ./fastlane/lib/fastlane/actions/create_pull_request.rb:18:in `run'
     # ./fastlane/lib/fastlane/runner.rb:256:in `block (2 levels) in execute_action'
     # ./fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
     # ./fastlane/lib/fastlane/runner.rb:235:in `block in execute_action'
     # ./fastlane/lib/fastlane/runner.rb:227:in `chdir'
     # ./fastlane/lib/fastlane/runner.rb:227:in `execute_action'
     # ./fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
     # ./fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
     # (eval):3:in `block in parsing_binding'
     # ./fastlane/lib/fastlane/lane.rb:33:in `call'
     # ./fastlane/lib/fastlane/runner.rb:49:in `block in execute'
     # ./fastlane/lib/fastlane/runner.rb:45:in `chdir'
     # ./fastlane/lib/fastlane/runner.rb:45:in `execute'
     # ./fastlane/spec/actions_specs/create_pull_request_spec.rb:64:in `block (5 levels) in <top (required)>'

  3) Fastlane Fastlane::FastFile create_pull_request successful correctly submits to github with assignees
     Failure/Error:
       result = Fastlane::FastFile.new.parse("
         lane :test do
           create_pull_request(
             api_token: '123456789',
             title: 'test PR',
             repo: 'fastlane/fastlane',
             assignees: ['octocat','hubot','other_user']
           )
         end
       ").runner.execute(:test)
     
     WebMock::NetConnectNotAllowedError:
       Real HTTP connections are disabled. Unregistered request: POST https://api.github.com/repos/fastlane/fastlane/pulls with body '{"title":"test PR","head":null,"base":"master"}' with headers {'Accept'=>'*/*', 'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'}
     
       You can stub this request with the following snippet:
     
       stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/pulls").
         with(:body => "{\"title\":\"test PR\",\"head\":null,\"base\":\"master\"}",
              :headers => {'Accept'=>'*/*', 'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'}).
         to_return(:status => 200, :body => "", :headers => {})
     
       registered request stubs:
     
       stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/issues/1347/assignees").
         with(:body => "{\"assignees\":[\"octocat\",\"hubot\",\"other_user\"]}",
              :headers => {'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'})
       stub_request(:patch, "https://api.github.com/repos/fastlane/fastlane/issues/1347").
         with(:body => "{\"labels\":[\"fastlane\",\"is\",\"awesome\"]}",
              :headers => {'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'})
       stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/pulls").
         with(:body => "{\"title\":\"test PR\",\"head\":\"git rev-parse --abbrev-ref HEAD\",\"base\":\"master\"}",
              :headers => {'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'})
       stub_request(:any, "/developer.apple.com\/services-account\/QH65B2\/account\/ios\/identifiers\/.*OMC(s){0,1}\.action/")
       stub_request(:any, "/developer.apple.com\/services-account\/QH65B2\/account\/auth\/key/")
       stub_request(:any, "/appstoreconnect.apple.com\/testflight\/v2/")
     
       ============================================================
     # /Library/Ruby/Gems/2.3.0/gems/webmock-2.3.2/lib/webmock/http_lib_adapters/excon_adapter.rb:68:in `handle_request'
     # /Library/Ruby/Gems/2.3.0/gems/webmock-2.3.2/lib/webmock/http_lib_adapters/excon_adapter.rb:40:in `block in add_excon_stub'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/mock.rb:38:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/instrumentor.rb:34:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/idempotent.rb:19:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/base.rb:22:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/middlewares/base.rb:22:in `request_call'
     # /Library/Ruby/Gems/2.3.0/gems/excon-0.64.0/lib/excon/connection.rb:270:in `request'
     # ./fastlane/lib/fastlane/actions/github_api.rb:250:in `call_endpoint'
     # ./fastlane/lib/fastlane/actions/github_api.rb:21:in `run'
     # ./fastlane/lib/fastlane/actions/create_pull_request.rb:18:in `run'
     # ./fastlane/lib/fastlane/runner.rb:256:in `block (2 levels) in execute_action'
     # ./fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
     # ./fastlane/lib/fastlane/runner.rb:235:in `block in execute_action'
     # ./fastlane/lib/fastlane/runner.rb:227:in `chdir'
     # ./fastlane/lib/fastlane/runner.rb:227:in `execute_action'
     # ./fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
     # ./fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
     # (eval):3:in `block in parsing_binding'
     # ./fastlane/lib/fastlane/lane.rb:33:in `call'
     # ./fastlane/lib/fastlane/runner.rb:49:in `block in execute'
     # ./fastlane/lib/fastlane/runner.rb:45:in `chdir'
     # ./fastlane/lib/fastlane/runner.rb:45:in `execute'
     # ./fastlane/spec/actions_specs/create_pull_request_spec.rb:79:in `block (5 levels) in <top (required)>'

  4) Fastlane::Actions::GitBranchAction with no CI set ENV values gets the value from Git directly
     Failure/Error: expect(result).to eq("branch-name")
     
       expected: "branch-name"
            got: nil
     
       (compared using ==)
     # ./fastlane/spec/actions_specs/git_branch_spec.rb:26:in `block (3 levels) in <top (required)>'

  5) Fastlane Fastlane::FastFile Slack Action parses default_payloads without adding extra fields for git
     Failure/Error: expect(fields.count).to eq(2)
     
       expected: 2
            got: 1
     
       (compared using ==)
     # ./fastlane/spec/actions_specs/slack_spec.rb:163:in `block (4 levels) in <top (required)>'

Finished in 4 minutes 10.3 seconds (files took 3.6 seconds to load)
5861 examples, 5 failures

Failed examples:

rspec ./fastlane/spec/actions_specs/create_pull_request_spec.rb:40 # Fastlane Fastlane::FastFile create_pull_request successful correctly submits to github
rspec ./fastlane/spec/actions_specs/create_pull_request_spec.rb:54 # Fastlane Fastlane::FastFile create_pull_request successful correctly submits to github with labels
rspec ./fastlane/spec/actions_specs/create_pull_request_spec.rb:69 # Fastlane Fastlane::FastFile create_pull_request successful correctly submits to github with assignees
rspec ./fastlane/spec/actions_specs/git_branch_spec.rb:17 # Fastlane::Actions::GitBranchAction with no CI set ENV values gets the value from Git directly
rspec ./fastlane/spec/actions_specs/slack_spec.rb:146 # Fastlane Fastlane::FastFile Slack Action parses default_payloads without adding extra fields for git

rake aborted!
Command failed with status (1): [rspec --pattern ./**/*_spec.rb --format pr...]
/Data/GitHub/fastlane/Rakefile:19:in `block in <top (required)>'
/Library/Ruby/Gems/2.3.0/gems/rake-11.3.0/exe/rake:27:in `<top (required)>'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:74:in `load'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:74:in `kernel_load'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/cli/exec.rb:28:in `run'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/cli.rb:463:in `exec'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/cli.rb:27:in `dispatch'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/cli.rb:18:in `start'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/exe/bundle:30:in `block in <top (required)>'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/lib/bundler/friendly_errors.rb:124:in `with_friendly_errors'
/Library/Ruby/Gems/2.3.0/gems/bundler-2.0.1/exe/bundle:22:in `<top (required)>'
/usr/local/bin/bundle:22:in `load'
/usr/local/bin/bundle:22:in `<main>'
Tasks: TOP => test_all
(See full trace by running task with --trace)
```

**Comment by arri-cc** [comment]
I think i'll await some advice/guidance before I make anymore changes.  I'm not sure what the rules are as it relates to dependencies between actions.

**Comment by janpio** [comment]
Re the failing tests: I think this is internally using `git_branch`, which is returning `null` now instead of the executed command which is normally returned when using `sh` during tests:

```
       stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/pulls").
         with(:body => "{\"title\":\"test PR\",\"head\":null,\"base\":\"master\"}",
              :headers => {'Accept'=>'*/*', 'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'}).
         to_return(:status => 200, :body => "", :headers => {})
```

vs.

```
  stub_request(:post, "https://api.github.com/repos/fastlane/fastlane/pulls").
         with(:body => "{\"title\":\"test PR\",\"head\":\"git rev-parse --abbrev-ref HEAD\",\"base\":\"master\"}",
              :headers => {'Authorization'=>'Basic MTIzNDU2Nzg5', 'Host'=>'api.github.com:443', 'User-Agent'=>'fastlane-github_api'})
```

Note `git rev-parse --abbrev-ref HEAD` in there.

If you change `      return ENV.fetch(env_name.to_s) { `git symbolic-ref HEAD --short 2>/dev/null`.strip }` to `      return ENV.fetch(env_name.to_s) { Actions.sh("git symbolic-ref HEAD --short 2>/dev/null", log: false).chomp` this would probably make sure that the command is returned - although it still might be a different one.

Do `git rev-parse --abbrev-ref HEAD` and `git symbolic-ref HEAD --short 2>/dev/null` the same thing?

**Comment by janpio** [comment]
> I'm not sure what the rules are as it relates to dependencies between actions.

Same here - this is probably also the reason for the helper. If they all use the helper, they are not interconnected but use some common code. We have this for some other bits as well I guess.

**Comment by arri-cc** [comment]
@janpio ok I'll take a closer look tonight, but if the right approach is to enhance the helper then the `git_branch` spec needs to be rewritten

**Comment by arri-cc** [comment]
> Do git rev-parse --abbrev-ref HEAD and git symbolic-ref HEAD --short 2>/dev/null the same thing?

The latter resolves to HEAD if no named branch is available instead of throwing an error.  I should use it instead.  The behavior is what exposed the differences.  The expanded set of environment variable names in the helper should resolve the original problem.

**Comment by janpio** [comment]
@joshdholtz Some guidance on the architecture of action vs. helper here please (we talked about this in other contexts recently).

**Comment by arri-cc** [comment]
As a workaround to this issue, instead of using the `ensure_git_branch` action, which will not get the correct branch name on non-jenkins CI servers, I just do this:
```ruby
branch = git_branch()
# alternate method of enforcing master branch
unless branch =~ /(?i)^(chore|feat|fix|refactor)\/(\w+-\d{2,3})(-\w+)*(?-i)/
  UI.user_error("branch does not match spec, current branch #{branch}")
end
```

Would like to know if I should continue down the path of consolidating use of the additional environment variables into the git helper or do something else.

File: fastlane/lib/fastlane/helper/git_helper.rb [hunk 1]
```diff
@@ -112,10 +116,8 @@ def self.last_git_commit_hash(short)
 
     # Returns the current git branch - can be replaced using the environment variable `GIT_BRANCH`
     def self.git_branch
-      return ENV['GIT_BRANCH'] if ENV['GIT_BRANCH'].to_s.length > 0 # set by Jenkins
-      s = Actions.sh("git rev-parse --abbrev-ref HEAD", log: false).chomp
-      return s.to_s.strip if s.to_s.length > 0
-      nil
+      env_name = SharedValues::GIT_BRANCH_ENV_VARS.find { |env_var| FastlaneCore::Env.truthy?(env_var) }
+      return ENV.fetch(env_name.to_s) { `git symbolic-ref HEAD --short 2>/dev/null`.strip }
```
Comment (dotdoom) [inline fastlane/lib/fastlane/helper/git_helper.rb]: `2>/dev/null` is not going to work well on Windows

**Comment by janpio** [comment]
@googlebot I fixed it.

**Pull request closed without merging**